lib/glific/templates.ex

```diff
@@ -449,25 +449,22 @@ defmodule Glific.Templates do
   @spec do_update_hsm(map(), map()) ::
           {:ok, SessionTemplate.t()} | {:error, Ecto.Changeset.t()}
   defp do_update_hsm(template, db_templates) do
-    current_template = db_templates[template["bsp_id"]]
-    update_attrs = %{status: template["status"]}
-
-    update_attrs =
-      if current_template.status != template["status"],
-        do:
-          Map.put(
-            update_attrs,
-            :is_active,
-            template["status"] in ["APPROVED"]
-          ),
-        else: update_attrs
-
-    {:ok, _} =
-      db_templates[template["bsp_id"]]
-      |> SessionTemplate.changeset(update_attrs)
-      |> Repo.update()
+    update_attrs = do_update_attrs(template["status"], template)
+
+    db_templates[template["bsp_id"]]
+    |> SessionTemplate.changeset(update_attrs)
+    |> Repo.update()
   end
 
+  @spec do_update_attrs(String.t(), map()) :: map()
```

Comment:
@AkhileshNegi if somebody deactivates the approved template, it will activate that again.

Can we fix it?

Reply:
@pankaj-ag Thanks for catching it. I have made changes to fix same and tested as well.
Assigning you for review again😁